### ISSUE TITLE
ocr prediction is too bad, model converge fast

### ISSUE THREAD

**Issue by pp3993** [emedvedev/attention-ocr#129]
Hi 
I have trained the model on 40,000 synthetic images generated [from](https://github.com/Belval/TextRecognitionDataGenerator). 
here is the modify CHARMAP:
CHARMAP = ['', '', ''] + ['-', 's', 'W', 'd', '.', 'Q', '[', '°', 'y', 'O', '©', 'Y', 'b', '/', '9', "'", 'H', 'V', 'z', ',', 'g', 'm', 'u', 'j', '>', 'S', 'c', ')', '@', 'k', 'q', '%', ':', 'L', 'o', 'E', '`', 'F', '$', '\\', '?', 'e', '¢', '~', 'f', 'G', '*', '•', 'p', '3', 'x', 'n', 'M', '_', '(', '€', '+', 'U', 'A', '#', 'h', '&', '=', 'K', '8', '7', 'w', '}', '6', 't', '0', '★', 'I', '|', 'v', 'i', ' ', 'C', 'P', '2', 'É', '{', 'r', '5', 'a', 'J', '¡', 'R', 'l', '£', '4', '!', 'T', 'Z', 'B', 'N', '^', '<', 'X', ']', 'D', ';', '"', '1']

Also, the change in parameters values are in bold letters:
    STEPS_PER_CHECKPOINT = 100
    **FORCE_UPPERCASE = False**
    SAVE_FILENAME = False
    FULL_ASCII = False
    NUM_EPOCH = 1000
    BATCH_SIZE = 65
    INITIAL_LEARNING_RATE = 1.0
    CLIP_GRADIENTS = True  # whether to perform gradient clipping
    MAX_GRADIENT_NORM = 5.0  # Clip gradients to this norm
    TARGET_EMBEDDING_SIZE = 10  # embedding dimension for each target
    ATTN_NUM_HIDDEN = 128  # number of hidden units in attention decoder cell
    ATTN_NUM_LAYERS = 2  # number of layers in attention decoder cell
    # (Encoder number of hidden units will be ATTN_NUM_HIDDEN*ATTN_NUM_LAYERS)
    LOAD_MODEL = True
    OLD_MODEL_VERSION = False
    TARGET_VOCAB_SIZE = 26+10+3  # 0: PADDING, 1: GO, 2: EOS, >2: 0-9, a-z
    CHANNELS = 1  # number of color channels from source image (1 = grayscale, 3 = rgb) 
    **MAX_WIDTH = 600 (max width of images are upto 565 pixel)**
    **MAX_HEIGHT = 100**
    **MAX_PREDICTION = 50**
    USE_DISTANCE = True
    LOG_STEP = 500

this is the train command and screenshot during training:

`aocr train /path/to/attention-ocr-master/aocr/datasets/training.tfrecords`

![train](https://user-images.githubusercontent.com/44362142/55556897-42951e80-5706-11e9-928d-086e69e8f494.jpg)

this is the test command and screenshot of testing with accuracy and text prediction:

`aocr test /path/to/attention-ocr-master/aocr/datasets/testing.tfrecords`

![test](https://user-images.githubusercontent.com/44362142/55556906-47f26900-5706-11e9-933f-753fbb8958be.jpg)

The training model converges very fast, but in testing it failed almost. Also, after exporting frozen graph predictions are not good.
please, can you help? why it is giving poor prediction? any changes required?

**Comment by pp3993** [comment]
Hey
I tried this again but no change in the prediction. Can you help ? please

**Comment by emedvedev** [comment]
Hi! I cannot help with the training, unless there's an actual error with the code. There's too much variables involved, both with your parameters and your dataset, and I just can't get too deep into it, sadly.

I'm going to close the issue, but if you find something in the code or the model itself that causes poor results, please feel free to report it.

**Comment by kunalgoyal9** [comment]
how many images needed for good prediction
